frontend/src/io-managers/input.ts

```diff
@@ -31,6 +31,13 @@ export function createInputManager(editor: Editor, container: HTMLElement, dialo
 
 	let viewportPointerInteractionOngoing = false;
 	let textInput = undefined as undefined | HTMLDivElement;
+	let inCanvas = true;
+
+	const app = window.document.getElementById("app");
```

Comment:
Nitpick: we have a slight preference to use `document.querySelector("#app")` for consistency, since we don't use the older `getElementById` JS API anywhere else in the codebase. If we ever need to search for all the functions in the project that interact with the DOM, it's nice to have them only be the three: `querySelector`, `querySelectorAll`, and `closest`.

Reply:
Also, this is another case where it would be slightly preferable to select based on a data attribute not a class name or ID, just to be clean about our separation between ID/class/tag names being for CSS only and data attributes being for JS only.